examples/dtls-sock/dtls_client_credentials.h

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2024 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       PSK and RPK client credentials for the dtls-sock example.
+ *
+ * @author      Leandro Lanzieri <leandro.lanzieri@haw-hamburg.de
+ *
+ * @}
+ */
+
+#ifndef DTLS_CLIENT_CREDENTIALS_H
+#define DTLS_CLIENT_CREDENTIALS_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * Pre-Shared Key (PSK) credentials (client and server must have a copy)
+ */
+
+/**
+ * @brief The identity of the PSK 0
+ */
+#define CLIENT_PSK_IDENTITY_0 "Identity_0"
+
+/**
+ * @brief The PSK 0 itself.
+ */
+#define CLIENT_PSK_IDENTITY_0_KEY "secretPSK"
+
+/**
+ * @brief The identity hint of the PSK 0
+ */
+#define CLIENT_PSK_IDENTITY_0_HINT "Prefer_Id_0"
```

Comment:
Would the hint be something like a hostname / node ID or serial number?

Reply:
Sure, but since this is an example I think some comment of what this would be in a real-world use case would be valuable.

Reply:
> something like a hostname / node ID or serial number

This would somehow imply that the key is only to be used with a single client, this is not necessarily the case. Searching for examples I found this summary: https://stackoverflow.com/a/8753393/5633371 seems that many implementations send encoded information on how to derive the keys.

I think this hint is fine for the didactic purpose of the example: It's easily readable in code and when sniffing the handshake, and it describes its intention.